lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -1934,6 +1934,10 @@ void LifetimeChecker::updateInstructionForInitState(DIMemoryUse &Use) {
       assert(InitKind == IsInitialization);
       AI->setOwnershipQualifier(AssignOwnershipQualifier::Reinit);
     } else {
+      if (AI->hasEnclosingSelfAccess() && InitKind != IsInitialization &&
+          getSelfInitializedAtInst(AI) != DIKind::Yes) {
+        diagnose(Module, Inst->getLoc(), diag::self_closure_use_uninit);
```

Comment:
This should be using the same `shouldEmitError(Inst)` precondition as the rest of the diagnostics. Otherwise dead usages of `self` will show up.